src/map/entities/battleentity.cpp

```diff
@@ -2691,3 +2691,32 @@ uint16 CBattleEntity::GetBattleTargetID() const
 {
     return m_battleTarget;
 }
+
+bool CBattleEntity::hasEnmity() const
+{
+    // TODO: This check seems to always fail for pets?
+    if (PNotorietyContainer->hasEnmity())
+    {
+        return true;
+    }
+
+    bool isTargeted = false;
+
+    // TODO: this is bad but because of how super tanking is implemented there's not much we can do without a larger refactor
+    if (loc.zone)
+    {
+        // clang-format off
+        loc.zone->ForEachMob([&](CMobEntity* PMob)
```

Comment:
When we eventually have spatial partitioning, we can do that sort of iteration here

Reply:
I don't think that will work here, because theoretically you could sleep a mob and Flee 100' away and it would STILL be mad at you